stock_picking_product_generate_sequence/models/stock_move_line.py

```diff
@@ -0,0 +1,49 @@
+##############################################################################
+# Copyright (c) 2022 brain-tec AG (https://braintec.com)
+# All Right Reserved
+#
+# See LICENSE file for full licensing details.
+##############################################################################
+from re import findall as regex_findall
+from re import split as regex_split
+
+from odoo import models, fields, api
+
+
+class StockMoveLine(models.Model):
+    _inherit = 'stock.move.line'
+
+    def action_generate_product_serial(self):
+        self.ensure_one()
+        self.lot_id = self.env['stock.production.lot'].create({
+            'product_id': self.product_id.id,
+            'company_id': self.company_id.id,
+            'name': self._get_next_serial(self.company_id, self.product_id) or
+                    self.env['ir.sequence'].next_by_code('stock.picking.product')
+        })
+
+    @api.model
+    def _get_next_serial(self, company, product):
+        sequence = product.product_tmpl_id.serial_lot
+        if sequence:
+            splitted = regex_split(r'(\d+)', sequence)
+            prefix = splitted[0]
+            initial_number = int(splitted[1])
```

Comment:
[Comment] Do you guarantee the serial_lot has the right format to assure you will get as a result 2 items?

Reply:
good point. I added a pattern check to guarantee the right format. thx

Reply:
What pattern check did you add to guarantee the right format?
Can you add the link here?

Reply:
Ah ok, you check it in an onchange function.
Maybe we should check it (also) in a constrain to ensure that the format is also checked when we import products or manipulate it from the code.
The onchange is only triggered in the view.

Reply:
I think it's right. I have added a constraint